src/components/templates/AnimateMap/game/map/components/HoverableSpriteComponent.ts

```diff
@@ -1,3 +1,24 @@
 export class HoverableSpriteComponent {
-  constructor(public on?: Function, public off?: Function) {}
+  constructor(on?: Function, off?: Function, isCursorPointer: boolean = false) {
+    if (!isCursorPointer) {
+      this.on = on;
+      this.off = off;
+    } else {
+      this.on = on
+        ? () => {
+            document.body.style.cursor = "pointer";
+            on();
+          }
+        : on;
+      this.off = off
+        ? () => {
+            document.body.style.cursor = "auto";
+            off();
+          }
+        : off;
+    }
+  }
+
+  public on?: Function;
+  public off?: Function;
 }
```

Comment:
A bit of refactor to make intent clearer while trying to keep the original logic

```ts
export class HoverableSpriteComponent {
  public on?: () => void;
  public off?: () => void;

  constructor(
    on?: () => void,
    off?: () => void,
    isCursorPointer: boolean = true
  ) {
    if (!isCursorPointer) {
      this.on = on;
      this.off = off;
      return;
    }

    this.on = () => {
      if (!on) return;
      document.body.style.cursor = "pointer";
      on();
    };

    this.off = () => {
      if (!off) return;
      document.body.style.cursor = "auto";
      off();
    };
  }
}
```

Reply:
Made it almost exactly as suggested in https://github.com/sparkletown/sparkle/pull/2161/commits/470154cca18bb5273e216b7d43acaace4ac25b4a